ml/synthesis/src/scripts/evaluate_nitta_synthesis.py

```diff
@@ -0,0 +1,378 @@
+"""
+A script for evaluation of NITTA's synthesis. It runs NITTA with different arguments, measures metrics (like time,
+synthesis success rate, etc.) and writes aggregated results to a single CSV table for further analysis.
+
+!!! Do not add non-stdlib imports here and be compatible with Python 3.8 !!!
```

Comment:
Oh, it'll actually be really easy to add a compatibility check to CI if we really want to enforce it!

A separate job with setup-python + 3.8, no dependencies installed, we don't even need a real nitta binary (we can use `/usr/bin/true` or something, not really important).

Reply:
Don't think it is necessary.